sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1803,7 +1803,7 @@ object SQLConf {
         "satisfying a query. This optimization allows object serializers to avoid " +
         "executing unnecessary nested expressions.")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefault(true)
```

Comment:
Hm, these two `spark.sql.optimizer.serializer.nestedSchemaPruning.enabled` and `spark.sql.optimizer.expression.nestedPruning.enabled` were added as of Spark 3.0 (SPARK-26837 and SPARK-27707). I thought it's rather usual to have one minor release term.

Reply:
Yes, this is added as part of Spark 3.0, but on really early stage of Spark 3.0 development. We internally cherry picked them into 2.4.x in our production Spark distributions, and those help a lot in many nested column use-cases.